modules/indexer/stats/db.go

```diff
@@ -37,6 +37,9 @@ func (db *DBIndexer) Index(id int64) error {
 
 	gitRepo, err := git.OpenRepositoryCtx(ctx, repo.RepoPath())
 	if err != nil {
+		if err.Error() == "no such file or directory" {
+			return nil
+		}
```

Comment:
Can we not compare against `os.ErrNotExist`?

Reply:
I think we are comparing against https://github.com/zeripath/gitea/blob/d3bf78ba3c67c5ed7beceb886a3f2540dc816c4e/modules/git/repo_base_nogogit.go#L50

That being said, comparing to error strings seems brittle. Can we instead export a concrete error to check against?

Reply:
I think the error string is incorrect too, as I think the meaning is there is no repo directory at the given path. I think we should create a custom error for this.

Reply:
This error string is chosen to match the error string returned by go-git

Reply:
Ok, in that case our hands are bound I guess.